src/View/Widget/DateTimeWidget.php

```diff
@@ -218,11 +218,11 @@ protected function _deconstructDate($value, $options)
             ];
         }
         try {
-            if (is_string($value)) {
+            if (is_string($value) && !is_numeric($value)) {
                 $date = new DateTime($value);
             } elseif (is_bool($value)) {
                 $date = new DateTime();
-            } elseif (is_int($value)) {
+            } elseif (is_int($value) || is_numeric($value)) {
                 $date = new DateTime('@' . $value);
```

Comment:
I think, insert timezone second parameter on new DateTime()
like this
```php
$date = new DateTime('@' . $value, $timezone);
```

but I know there are no timezone data here, too.